### DISCUSSION TITLE
Cannot find module  \node_modules\wdio\bin\wdio.js, Not getting installed properly

### DISCUSSION THREAD

**Discussion by anandseshadri** [webdriverio/webdriverio#8310] [Q&A]
Hi I am getting the error when i run the command `wdio wdio.conf.js --suite login`
`Error: Cannot find module 'AppData\Local\Temp\dlx-21520\5\node_modules\wdio\bin\wdio.js'
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:77:12)
    at node:internal/main/run_main_module:17:47 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}`

I am not sure why the wdio is not getting installed properly

**Comment by christian-bromann** [comment]
> I am not sure why the wdio is not getting installed properly

Because you didn't install the CLI: `npm install @wdio/cli`. It seems you have installed the [`wdio`](https://www.npmjs.com/package/wdio) which is deprecated.

**Reply by anandseshadri** [reply]
I am getting the same error after installing CLI: `npm install @wdio/cli`

**Reply by christian-bromann** [reply]
If you install the package locally, make sure you call it via `npx wdio run wdio.conf.js`, see more https://www.npmjs.com/package/npx

**Reply by anandseshadri** [reply]
@christian-bromann  Thank you for your help, I did use the command `npm install @wdio/cli -g` still getting the error, I am not sure why its not getting installed globally, Using npx as you suggested resolved the issue.  I want to get it properly installed globally

**Reply by NamrathaVarma** [reply]
I am also getting the same error : Cannot find module 'D:\App Automation\wdio.conf.js' imported from D:\App Automation\node_modules\@wdio\config\build\lib\RequireLibrary.js

I also tried npm install @wdio/cli -g and doesnt seem to work, can I please get help on this?

**Reply by christian-bromann** [reply]
Can you provide a minimal reproducible example?

**Comment by NamrathaVarma** [comment]
I fixed by re-creating the whole thign again and ran these series of commands that helped me get my config file setup on VS code

mkdir webdriverio-test
cd webdriverio-test 
npm init -y 
npm i --save-dev @wdio/cli
npx wdio config -y 
mkdir .\test\specs
npx wdio wdio.conf.js

**Reply by Gusmatto** [reply]
Thanks, NamrathaVarma, you saved me!!!

**Reply by anbumaran** [reply]
Thank you, It worked for me!
>npm run wdio

**Comment by reemadaga** [comment]
@christian-bromann Any idea why I would be getting this error:
2024-01-11T06:45:43.029Z ERROR @wdio/config:ConfigParser: Failed loading configuration file: file:///Users/XXX/wdio.conf.js: Cannot find module '/XXX/src/configs/wdio.conf.js' imported from /XXX/node_modules/@wdio/config/build/node/RequireLibrary.js
node:internal/errors:496
    ErrorCaptureStackTrace(err);
    ^

Error [ERR_MODULE_NOT_FOUND]: Cannot find module 'XXX/src/configs/wdio.conf.js' imported from /XXX/node_modules/@wdio/config/build/node/RequireLibrary.js
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ESMLoader.import (node:internal/modules/esm/loader:524:22)
    at importModuleDynamically (node:internal/modules/esm/translators:110:35)
    at importModuleDynamicallyCallback (node:internal/process/esm_loader:36:14)
    at RequireLibrary.import 
  code: 'ERR_MODULE_NOT_FOUND'
}

Node.js v18.18.0

**Reply by christian-bromann** [reply]
well seems like `XXX/src/configs/wdio.conf.js` doesn't exist.

**Reply by reemadaga** [reply]
I was getting some error for html report so I added type:module to package.json and now my package is all messed up.

**Reply by reemadaga** [reply]
Instead of looking for wdio.conf.js in the "dist" folder it is looking under where .ts exists.

**Reply by erwinheitzman** [reply]
Sounds like you are mixing up typescript and JavaScript by first compiling to JavaScript, let wdio do the transpilation instead if that is the case

**Comment by reemadaga** [comment]
@erwinheitzman Can you please suggest the compiler options, sharing what I have right now:
"compilerOptions": {
        "outDir": "./dist",
        "target": "ES2020",
        "module": "commonjs",
        "moduleResolution": "node",
        "strict": true,
        "strictPropertyInitialization": false,
        "lib": ["es6", "dom"],
        "types": [
            "node", 
            "@wdio/globals/types",
            "@wdio/mocha-framework", 
            "wdio-native-app-compare-service",
            "expect-webdriverio",
            "./src/types/global",
        ],

**Reply by erwinheitzman** [reply]
You are compiling the code to JavaScript and outputting it to a dist folder. You should not do that, instead just run the typescript files as they are

**Comment by reemadaga** [comment]
@erwinheitzman  - Running typescript now and running into this issue:
esbuild Error]: Top-level await is currently not supported with the "cjs" output format

**Reply by erwinheitzman** [reply]
Is this a new/first project you have set up? Because this means that your typescript configuration isn't correct

**Reply by reemadaga** [reply]
@erwinheitzman -
I am upgrading from webdriverio 6 to 8. 
"compilerOptions": {
"outDir": "./dist",
"target": "ES2020",
"module": "commonjs",
"strict": true,
"strictPropertyInitialization": false,
"lib": ["es6", "dom"],
"types": [
"node",
"@wdio/globals/types",
"@wdio/mocha-framework",
"wdio-native-app-compare-service",
"expect-webdriverio",
"./src/types/global",
],
should I switch to target: ESNEXT and module: ESNEXT? I will have to update my entire package so was trying to see if what I have can work.

**Reply by erwinheitzman** [reply]
You can remove the outputDir as you are not outputting any JavaScript, you only want to transpile. You need to update the module to either Node16 or NodeNext iirc. The target does not matter and can be kept as it is. You might need to add the module moduleResolution which if needed probably needs to be set to node
